apps/masterbots.ai/lib/hooks/use-sidebar.tsx

```
@@ -163,25 +171,31 @@ export function SidebarProvider({ children }: SidebarProviderProps) {
-          .map(category => category.categoryId)]
-        : prev.filter(id => !categories?.categoriesChatbots
-          .filter(category => category.chatbots.some(chatbot => chatbot.chatbotId === chatbotId))
-          .map(category => category.categoryId)
-          .includes(id)
-        )
+        .filter((category) => category.chatbots.some((chatbot) => chatbot.chatbotId === chatbotId))
+        .map((category) => category.categoryId)
+        .filter((id) => !prev.includes(id)).length
+        ? [
+          ...prev,
+          ...categories?.categoriesChatbots
```

Biome (1.9.4)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)